api/api_ethereum.go

```diff
@@ -851,10 +945,18 @@ func newEthTransactionReceipt(tx *types.Transaction, blockHash common.Hash, bloc
 		"contractAddress":   nil,
 		"logs":              receipt.Logs,
 		"logsBloom":         receipt.Bloom,
-		"type":              hexutil.Uint(typeInt),
+		"type":              hexutil.Uint(byte(typeInt)),
 	}
 
-	fields["effectiveGasPrice"] = tx.GasPrice()
+	// TODO-Klaytn: Klaytn is using fixed BaseFee(0) as now but
+	// if we apply dynamic BaseFee, we should add calculated BaseFee instead of using params.BaseFee.
+	baseFee := new(big.Int).SetUint64(params.BaseFee)
+	fields["effectiveGasPrice"] = hexutil.Uint64(
+		new(big.Int).Add(
+			baseFee,
+			tx.EffectiveGasTip(baseFee),
+		).Uint64(),
+	)
```

Comment:
How about using `tx.EffectiveGasPrice(baseFee)`?

Reply:
Below is the full source code of Geth about creating value for "effectiveGasPrice" field.
```go
	if !s.b.ChainConfig().IsLondon(bigblock) {
		fields["effectiveGasPrice"] = hexutil.Uint64(tx.GasPrice().Uint64())
	} else {
		header, err := s.b.HeaderByHash(ctx, blockHash)
		if err != nil {
			return nil, err
		}
		gasPrice := new(big.Int).Add(header.BaseFee, tx.EffectiveGasTipValue(header.BaseFee))
		fields["effectiveGasPrice"] = hexutil.Uint64(gasPrice.Uint64())
	}
```

I think using same formula with Ethereum looks great but as you said,
baseFee is currently set to `0` and we can use that formula too.

Reply:
I checked the Ethereum's code, and I think it's not appropriate. How do you think?